example/node-fetch.ts

```diff
@@ -7,27 +7,22 @@ export async function callBackend({
 }: CallBackendArguments<{}>) {
   const searchPath = query ? `?${new URLSearchParams(query)}` : '';
   const url = `${baseUrl}${path}${searchPath}`;
+  switch (bodyType) {
+    case 'json':
+      headers = {...headers, 'content-type': 'application/json'};
```

Comment:
I prefer not to mutate the header and body types. If you want to improve the fetch example you could create some more `const` declarations to make it more readable.

Reply:
It's not a mutation. Here I change reference from one object to another, assign operator will do it in the local scope, original user provided arguments won't be mutated. If you want I can edit this.

Reply:
This rule covers this btw, consider adding it to your ESLint: https://eslint.org/docs/rules/no-param-reassign

Reply:
Fixed.